### ISSUE TITLE
Training loop

### ISSUE THREAD

**Issue by x33rak** [Ephemeral182/UDR-S2Former_deraining#8]
Hi, I enjoyed reading the paper and testing the project that you have developed. 

I trained on my end and noticed that as the train loss and validation loss keeps decreasing (becomes negative even) the PSNR and SSIM still continue to increase. 

When I train I got a train loss of -44.28 and validation loss of -29.18 but the PSNR is 27.73 and SSIM is .8729 (best case in my training). In my understanding shouldn't the loss be closer to 0 if you are trying to get a 1:1 correspondence between ground truth and the raindrop image.

**Comment by Ephemeral182** [comment]
Hi, because we utilize PSNR loss https://github.com/Ephemeral182/UDR-S2Former_deraining/blob/82101949f7549994eaf8aac2689b5c8fc5e8e080/loss/CL1.py#L40 
to supervise our training process, which means the lower the loss, the higher the PSNR.

In addition, many forms of loss function are negative, and they do not need to be close to 0. The backpropagation of the gradient is the most important.

**Comment by x33rak** [comment]
I see! Thank you for your help! I will look into the code further.
